src/media/UAudioPlaybackBase.pas

```diff
@@ -248,18 +265,40 @@ procedure TAudioPlaybackBase.Play;
 begin
   if assigned(MusicStream) then
     MusicStream.Play();
+  if assigned(KaraokeMusicStream) then
+    KaraokeMusicStream.Play();
 end;
 
 procedure TAudioPlaybackBase.Pause;
 begin
   if assigned(MusicStream) then
     MusicStream.Pause();
+  if assigned(KaraokeMusicStream) then
+    KaraokeMusicStream.Pause();
+end;
+
+procedure TAudioPlaybackBase.ToggleKaraoke;
+begin
+  KaraokeMode := not KaraokeMode;
+  if KaraokeMode and assigned(KaraokeMusicStream) and assigned(MusicStream) then begin
+    KaraokeMusicStream.Volume := MusicStream.Volume;
+    MusicStream.Volume := 0;
+    Log.LogStatus('Music now at zero!!!', 'TAudioPlayback_Base.ToggleKaraoke');
```

Comment:
Are these log statements necessary? (the second one is a few lines down). It's not the end of the world if you want to leave them in, but then at least take out the exclamation marks. Maybe even change it into a `LogDebug` as well but that's up to you, this does have a use if the instrumental is shorter.

Reply:
will remove the log statements